src/sdk/bucketspace_multi_fs.js

```diff
@@ -0,0 +1,399 @@
+/* Copyright (C) 2020 NooBaa */
+'use strict';
+
+const path = require('path');
+const config = require('../../config');
+const nb_native = require('../util/nb_native');
+const SensitiveString = require('../util/sensitive_string');
+const { S3Error } = require('../endpoint/s3/s3_errors');
+const RpcError = require('../rpc/rpc_error');
+const net = require('net');
+const js_utils = require('../util/js_utils');
+const P = require('../util/promise');
+const BucketSpaceFS = require('./bucketspace_fs');
+const _ = require('lodash');
+const dbg = require('../util/debug_module')(__filename);
+
+const VALID_BUCKET_NAME_REGEXP =
+    /^(([a-z0-9]|[a-z0-9][a-z0-9-]*[a-z0-9])\.)*([a-z0-9]|[a-z0-9][a-z0-9-]*[a-z0-9])$/;
+
+//TODO:  dup from namespace_fs - need to handle and not dup code
+function isDirectory(ent) {
+    if (!ent) throw new Error('isDirectory: ent is empty');
+    if (ent.mode) {
+        // eslint-disable-next-line no-bitwise
+        return (((ent.mode) & nb_native().fs.S_IFMT) === nb_native().fs.S_IFDIR);
+    } else if (ent.type) {
+        return ent.type === nb_native().fs.DT_DIR;
+    } else {
+        throw new Error(`isDirectory: ent ${ent} is not supported`);
+    }
+}
+
+
+class BucketSpaceMultiFS extends BucketSpaceFS {
+    constructor({fs_root, iam_dir, multi_fs_config_dir}) {
+        super({fs_root});
+        this.fs_root = fs_root;
+        this.iam_dir = iam_dir;
+        this.multi_fs_config_dir = multi_fs_config_dir;
+        this.fs_context = {
+            uid: process.getuid(),
+            gid: process.getgid(),
+            warn_threshold_ms: config.NSFS_WARN_THRESHOLD_MS,
+        };
+    }
+
+    async read_account_by_access_key({ access_key }) {
+        try {
+            console.log('GGG read_account_by_access_key', this.iam_dir, access_key);
+            if (!access_key) throw new Error('no access key');
+            const iam_path = path.join(this.iam_dir, access_key);
+            console.log('GGG read_account_by_access_key', access_key, iam_path);
+            const { data } = await nb_native().fs.readFile(this.fs_context, iam_path);
+            const account = JSON.parse(data.toString());
+            account.name = new SensitiveString(account.name);
+            account.email = new SensitiveString(account.email);
+            for (const k of account.access_keys) {
+                k.access_key = new SensitiveString(k.access_key);
+                k.secret_key = new SensitiveString(k.secret_key);
+            }
+            console.log('GGG read_account_by_access_key', access_key, account);
+            return account;
+        } catch (err) {
+            throw new RpcError('NO_SUCH_ACCOUNT', `Account with access_key not foundSSSSS`, err);
+        }
+    }
+
+    async read_bucket_sdk_info({ name }) {
+        const bucket_path = path.join(this.fs_root, name);
+        const { data } = await nb_native().fs.readFile(this.fs_context, bucket_path + ".json");
```

Comment:
fix bucket json path to use multi_fs_config_dir
```suggestion
        const bucket_config_path = path.join(this.multi_fs_config_dir, 'buckets', name + '.json');
        const { data } = await nb_native().fs.readFile(this.fs_context, bucket_config_path);
```